File: leetcode/3169. Count Days Without Meetings/main.ts
const countDays = (days: number, meetings: number[][]): number => {
  meetings.sort((a, b) => a[0] - b[0]);

  let merged: number[][] = [];

  for (let meet of meetings) {
    if (merged.length > 0 && merged[merged.length - 1][1] >= meet[0]) {
      merged[merged.length - 1][1] = Math.max(merged[merged.length - 1][1], meet[1]);
    } else {
      merged.push(meet);
    }
  }

  let result = 0;
  for (let meet of merged) {
    result += meet[1] - meet[0] + 1;
  }

  return days - result;
};

File: leetcode/3169. Count Days Without Meetings/main.rs
impl Solution {
    pub fn count_days(days: i32, mut meetings: Vec<Vec<i32>>) -> i32 {
        meetings.sort_by_key(|m| m[0]);

        let mut merged: Vec<Vec<i32>> = Vec::new();

        for meet in meetings {
            if let Some(last) = merged.last_mut() {
                if last[1] >= meet[0] {
                    last[1] = last[1].max(meet[1]);
                    continue;
                }
            }

            merged.push(meet)
        }

        let mut result = 0;
        for meet in &merged {
            result += meet[1] - meet[0] + 1
        }

        days - result
    }
}

